### PR TITLE
Uses built-in Android strings for base-ui

### PR DESCRIPTION
**Pull request by fstanis**
#### WHAT
Makes `base-ui` use [Android built-in strings](https://developer.android.com/reference/android/R.string).

#### WHY
Make translation easier.

#### HOW
By removing `res` all together from base-ui.

#### Checklist :clipboard:
- [X] Add explicit visibility modifier and explicit return types for public declarations
- [X] Run spotless check
- [X] Run tests
- [X] Update metalava's signature text files

### CODE REVIEW

**Review by luizgrp — APPROVED** [review]

File: media-ui/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -39,9 +39,9 @@
...
     <string description="Content description of the Download in Progress button, on the Playlist Download screen. It lets the user know that the playlist is currently being downloaded, and lets the user cancel the download of the playlist. [CHAR_LIMIT=NONE]" name="horologist_playlist_download_progress_button_cancel_content_description">Download in progress</string>
-    <string description="Description of the action of the Download in Progress button on the screen of a playlist that is currently being downloaded. Lets the user know that taking this action, it will remove the download. [CHAR_LIMIT=NONE]" name="horologist_playlist_download_progress_button_cancel_action_label">cancel</string>
+    <string description="Description of the action of the Download in Progress button on the screen of a playlist that is currently being downloaded. Lets the user know that taking this action, it will remove the download. [CHAR_LIMIT=NONE]" name="horologist_playlist_download_progress_button_cancel_action_label">Cancel</string>
```
Comment (laiyichin) [inline media-ui/src/main/res/values/strings.xml:42]: We are soon adding canceling functionality for individual media. Shall we rephrase this to something like "playlist/media"?

Comment (laiyichin) [reply media-ui/src/main/res/values/strings.xml:42]: Actually, I can do that along with adding the functionality.

Comment (luizgrp) [reply media-ui/src/main/res/values/strings.xml:42]: Given they are different buttons, I think it is better to have separate content descriptions instead of reusing them just because the text is the same

File: base-ui/src/main/java/com/google/android/horologist/base/ui/components/AlertDialog.kt [hunk 2]
```diff
@@ -43,8 +43,8 @@ public fun AlertDialog(
     scalingLazyListState: ScalingLazyListState,
     modifier: Modifier = Modifier,
     title: String = "",
-    okButtonContentDescription: String = stringResource(R.string.alert_dialog_ok_button_content_description),
-    cancelButtonContentDescription: String = stringResource(R.string.alert_dialog_cancel_button_content_description)
+    okButtonContentDescription: String = stringResource(R.string.ok),
```
[the remark this answers is not in the document]
Comment (fstanis) [reply base-ui/src/main/java/com/google/android/horologist/base/ui/components/AlertDialog.kt:46]: Yeah, I think using Android strings for defaults works, but we should do it sparingly.
